service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/model/common/Notification.java

```diff
@@ -58,13 +63,30 @@
 
     public static Notification fromKeyValueString(final String keyValueString, final String separatorPattern) {
 
-        Map<String, String> notificationValues = Splitter.onPattern(separatorPattern).withKeyValueSeparator("=").split(keyValueString);
+        final Map<String, String> notificationValues = Pattern.compile(separatorPattern)
+               .splitAsStream(keyValueString.trim())
+               .map(s -> s.split("="))
+               .collect(Collectors.toMap(a -> a[0], a -> a.length>1? a[1]: ""));
+
+        validateNotificationValues(notificationValues);
```

Comment:
https://github.com/commercetools/commercetools-payone-integration/pull/367#discussion_r612501179

Reply:
Just curious, do we need to validate it?

Reply:
Actually yes, otherwise the Notification object will not be properly initialized. But i recommend to use a different approach to parse the requestBody into Notification object. This is pretty complex and its pretty hard to read...

Reply:
The question is what will happen if we throw an exception here? What response code will we return to payone? In case it is not 200 then Payone will retry with the same wrong payload multiple times later and this single notification might block other / valid notifications too. In case a notification is not valid and can not be processed on our end we should instead log an error and return OK/200 to Payone.

Reply:
Current implementation logs the Error and responses with `400`: https://github.com/commercetools/commercetools-payone-integration/blob/53a72392a8b1e9e3817e38887a617cf84f6c9dc4/service/src/main/java/com/commercetools/pspadapter/payone/IntegrationService.java#L106
If there's a need to change please create a ticket.

Reply:
Thanks, ticket created.